src/Bicep.Cli/Helpers/VersionChecker.cs

```diff
@@ -0,0 +1,228 @@
+            {
+                locations.Add(fileSystem.Path.Combine(programFiles, "Bicep CLI"));
+            }
+
+            // C:\Program Files (x86)\Bicep CLI
+            var programFilesX86 = environment.GetVariable("ProgramFiles(x86)");
+            if (!string.IsNullOrEmpty(programFilesX86))
+            {
+                locations.Add(fileSystem.Path.Combine(programFilesX86, "Bicep CLI"));
+            }
+
+            var pathVar = environment.GetVariable("PATH");
+            if (!string.IsNullOrEmpty(pathVar))
+            {
+                foreach (var pathEntry in pathVar.Split(';'))
+                {
+                    if (!string.IsNullOrWhiteSpace(pathEntry))
+                    {
+                        locations.Add(pathEntry.Trim());
+                    }
+                }
+            }
+        }
+        else
+        {
+            // /usr/local/bin
+            locations.Add("/usr/local/bin");
+
+            // /usr/bin
+            locations.Add("/usr/bin");
+
+            var pathVar = environment.GetVariable("PATH");
+            if (!string.IsNullOrEmpty(pathVar))
+            {
+                foreach (var pathEntry in pathVar.Split(':'))
+                {
+                    if (!string.IsNullOrWhiteSpace(pathEntry))
+                    {
+                        locations.Add(pathEntry.Trim());
+                    }
+                }
+            }
+        }
+
+        return locations.Distinct(StringComparer.OrdinalIgnoreCase).ToList();
+    }
+
+    private Version? GetBicepVersion(string bicepPath)
+    {
+        try
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = bicepPath,
+                Arguments = "--version",
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                UseShellExecute = false,
+                CreateNoWindow = true
+            };
+
+            using var process = Process.Start(startInfo);
+            if (process == null)
+            {
+                return null;
+            }
+
+            process.WaitForExit(5000);
+
+            if (process.ExitCode != 0)
+            {
+                return null;
+            }
+
+            var output = process.StandardOutput.ReadToEnd();
+
+            // Parse version from output like "Bicep CLI version 0.30.23 (abc123)"
+            // Extract the version number between "version " and the next space or parenthesis
+            var versionMatch = System.Text.RegularExpressions.Regex.Match(
+                output,
+                @"version\s+(\d+\.\d+\.\d+)",
+                System.Text.RegularExpressions.RegexOptions.IgnoreCase);
+
+            if (versionMatch.Success && Version.TryParse(versionMatch.Groups[1].Value, out var version))
+            {
+                return version;
+            }
+
+            return null;
+        }
+        catch
+        {
+            return null;
+        }
```

Comment:
Could we consider using [`FileVersionInfo`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.fileversioninfo?view=net-9.0) here, rather than launching a process and parsing the output? i.e.

```csharp
var fileVersionInfo = System.Diagnostics.FileVersionInfo.GetVersionInfo(bicepPath);
var version = fileVersionInfo.FileVersion;
```

Reply:
agreed and done, thanks 👍